src/io.ts

```diff
@@ -1,9 +1,26 @@
 import { createHash } from 'crypto';
 import { join, parse } from 'path';
+import { promisify } from 'util';
+import { access } from 'fs';
 import { ensureDir, readdir, readJSON, unlink, writeJson } from 'fs-extra';
 import { buildTodoData } from './builders';
 import { FilePath, LintResult, TodoData } from './types';
 
+const exists = promisify(access);
+
+export async function todoDirExists(baseDir: string): Promise<boolean> {
```

Comment:
Hmm. I don't fully grok the point of this API. I thought we were going to let the caller of our APIs choose what directory to use? If that is the case, how can we actually do this?

Reply:
If we go down the path of letting the caller define the dir, then it's not needed, but I'm not 100% sure we have decided on that. I know we started to look at implementing that in `ember-template-lint`, but I'm not convinced there's a point to it. Why do we need them to specify it?

I see this type of directory as similar to what other tools use as storage directories, and it just adds unnecessary complexity to the solution IMO, but am open to being convinced otherwise.

Reply:
If the dir is totally opaque and never modified I think it makes sense, but IMHO if we intend to add non-automatically generated info to the file (e.g. due dates) then it seems like it really needs to be in the control of the host app with a sensible default.

Reply:
I was thinking that with due dates we'd allow the user to specify them when they generate/regenerate the files. These would be the cases:

1. No generated files, pass in due date when first generating
2. Existing generated files, pass in due date, will add due date to new todos _and_ those without an already defined due date
3. Existing generated files, pass in due date, all due dates already defined, noop.

The above is orthogonal to this discussion, but I also think due dates are also. Why does the presence of due dates require control of the storage directory location? Is it because the todos would contain custom, non-reproducible data? Why does the presence of a custom path for the storage directory help with this? You could still delete that custom directory, but I do concede you could 'hide' it better so it's less likely to be deleted/modified.

I'm interested in your rationale more. I'm open to being convinced, but right now it does seem like extra complexity.

Reply:
Chatted about this offline, we can tackle this later...